turbinia/processors/archive.py

```diff
@@ -0,0 +1,118 @@
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""File archiving processor"""
+
+from __future__ import unicode_literals
+
+import os
+import tarfile
+import logging
+
+from time import time
+from turbinia import TurbiniaException
+
+log = logging.getLogger('turbinia')
+
+
+def ValidateTarFile(compressed_directory):
+  """ Validates a given compressed directory path.
+
+  Performs a check to determine if the path exists and if
+  the file extension is in the list of accepted extensions.
+
+  Args:
+    compressed_directory(str): The path to the compressed tar file.
+
+  Raises:
+    TurbiniaException: If validation fails.
+  """
+  if not os.path.exists(compressed_directory):
+    raise TurbiniaException(
+        'The File or Directory does not exist: {0:s}'.format(
+            compressed_directory))
+
+  # TODO(wyassine): rewrite this check so it is not dependant
+  # on a list of hard coded extensions and instead have a
+  # check to determine whether or not it is a tar file format.
+  if not (compressed_directory.endswith('.tgz') or
+          compressed_directory.endswith('.tar.gz')):
+    raise TurbiniaException(
+        'The file is not a supported format. The list of '
+        'acceptable exensions are: .tgz or .tar.gz')
+
+
+def CompressDirectory(uncompressed_directory):
+  """Compress a given directory into a tar file.
+
+  Args:
+    uncompressed_directory(str): The path to the uncompressed directory.
+
+  Returns:
+    str: The path to the tar file.
+  """
+  # Error handling check for a non-existent file or directory.
+  if not os.path.exists(uncompressed_directory):
+    raise TurbiniaException(
+        'The File or Directory does not exist: {0:s}'.format(
+            uncompressed_directory))
+
+  # Iterate through a given list of files and compress them.
+  compressed_directory = uncompressed_directory + '.tar.gz'
+  try:
+    with tarfile.TarFile.open(compressed_directory, 'w:gz') as tar:
+      tar.add(uncompressed_directory, arcname='')
+      tar.close()
+      log.info(
+          'The tar file has been created and '
+          'can be found at: {0:s}'.format(compressed_directory))
+  except IOError as exception:
+    raise TurbiniaException('An error has occured: {0:s}'.format(exception))
+  except tarfile.TarError as exception:
+    raise TurbiniaException(
+        'An error has while compressing the directory: {0:s}'.format(exception))
+  return compressed_directory
+
+
+def UncompressTarFile(compressed_directory, output_tmp):
+  """Uncompress a provided tar file.
+
+  Args:
+    compressed_directory(str): The path to the tar file.
```

Comment:
Can you add `output_tmp` to the docstring Args here?

Reply:
Fixed.